R/4_standard_error.R

```diff
@@ -135,6 +157,77 @@ standard_error.default <- function(model, method = NULL, verbose = TRUE, ...) {
 }
 
 
+.get_vcov <- function(x,
+                      vcov_fun = "vcovHC",
+                      vcov_args = NULL,
+                      component = "conditional",
+                      method = "any",
+                      ...) {
+
+  dots <- list(...)
+
+  if (is.null(vcov_args)) {
+    vcov_args <- list()
+  }
+
+  # deprecated: `vcov_estimation`
+  if (is.null(vcov_fun) && "vcov_estimation" %in% names(dots)) {
+    vcov_fun <- dots[["vcov_estimation"]]
+  }
+
+  # deprecated: `robust`
+  if (isTRUE(dots[["robust"]]) && is.null(vcov_fun)) {
+    dots[["robust"]] <- NULL
+    vcov_fun <- "HC3"
+  }
+
+  # deprecated: `vcov_type`
+  if ("vcov_type" %in% names(dots)) {
+    if (!"type" %in% names(vcov_args)) {
+      vcov_args[["type"]] <- dots[["vcov_type"]]
+    }
+  }
+
+  # type shortcuts: overwrite only if not supplied explicitly by the user
+  if (vcov_fun %in% c("HC0", "HC1", "HC2", "HC3", "HC4", "HC4m", "HC5")) {
```

Comment:
Let's use switch() here instead eg,

```suggestion
  switch(vcov_fun,  HC0 =, HC1 =, HC2 =, HC3 =, HC4 =, HC5 = ...
```